history.go

```diff
@@ -7,37 +7,60 @@ import (
 )
 
 var (
-	ErrBadEpoch    = errors.New("[merkletree] Next epoch's STR has bad epoch")
-	ErrSTRNotFound = errors.New("[merkletree] STR not found")
+	ErrBadEpoch           = errors.New("[merkletree] Next epoch's STR has bad epoch")
+	ErrSTRNotFound        = errors.New("[merkletree] STR not found")
+	ErrMalformedHashChain = errors.New("[merkletree] Malformed hash chain")
 )
 
+type Epoch int64
+
 type History struct {
-	str           *SignedTreeRoot
-	epochInterval int64
+	currentSTR    *SignedTreeRoot
+	hashChain     map[Epoch]*SignedTreeRoot
+	epochs        []Epoch // slice of epochs in hashChain
+	epochInterval Epoch
```

Comment:
> I think it is quicker when accessing this field in the History struct, instead of referencing to the value stored in the STR's policy.

The reason why I prefer to not have a value stored in two places is because we then always have to ensure that they are consistent or synchronized with each other. If the policy is accessible to the ``History`` struct via the STR, even if it's a bit slower, I think it's safer and anybody reading the code knows exactly where the value came from.